apps/sms/views/shared/js/smil.js

```diff
@@ -365,9 +369,10 @@ window.SMIL = {
     if (!smil || attachmentsNotFound || !slides.length || smilMismatched) {
       // reset slides in the attachments not found case
       slides = Array(attachments.length);
-      attachments.forEach(SMIL_parseWithoutSMIL);
+      result = Promise.all(attachments.map(SMIL_parseWithoutSMIL));
     }
-    exitPoint();
+
+    return (result || Promise.resolve()).then(() => slides);
```

Comment:
I know I could have done better, especially a better use of Promise.reject to handle errors and fallbacks. As a first step I wanted to get the API right and update unit tests, so that I can use a Promise API in the new code. I can file a separate bug to make this better if you think it's worth it?